modules/aspect_bazel_lib/0.4.0/patches/extensions.patch

```diff
@@ -0,0 +1,125 @@
+commit 341024e64baf78032253aa04ffc3982f5b2e4d3b
+Author: Alex Eagle <alex@aspect.dev>
+Date:   Tue Dec 14 19:57:02 2021 -0800
+
+    feat: add bzlmod extension
+
+diff --git a/lib/extensions.bzl b/lib/extensions.bzl
+new file mode 100644
+index 0000000..f2aa62e
+--- /dev/null
++++ b/lib/extensions.bzl
+@@ -0,0 +1,44 @@
++"extensions for bzlmod"
++
++load("//lib/private:symlink_repository.bzl", "symlink_repository")
++load(":repositories.bzl", "register_jq_toolchains")
++
++jq_toolchain = tag_class(attrs = {
++    "name": attr.string(doc = "Base name for generated repositories"),
++    "version": attr.string(doc = "Version of the jq tool"),
++})
++
++def _toolchain_extension(module_ctx):
++    registrations = {}
++    for mod in module_ctx.modules:
++        for toolchain in mod.tags.jq_toolchain:
++            if toolchain.name in registrations.keys():
++                if toolchain.version == registrations[toolchain.name]:
++                    # No problem to register a matching toolchain twice
++                    continue
++                fail("Multiple conflicting toolchains declared for name {} ({} and {})".format(
++                    toolchain.name,
++                    toolchain.version,
++                    registrations[toolchain.name],
++                ))
++            else:
++                registrations[toolchain.name] = toolchain.version
++    for name, version in registrations.items():
++        register_jq_toolchains(
++            name = name,
++            version = version,
++            register = False,
++        )
++
++ext = module_extension(
++    implementation = _toolchain_extension,
++    tag_classes = {"jq_toolchain": jq_toolchain},
++)
++
++def _test_repos_impl(module_ctx):
++    symlink_repository(
++        name = "external_test_repo",
++        path = "lib/tests/external_test_repo",
++    )
++
++test_repos = module_extension(_test_repos_impl)
+diff --git a/lib/private/symlink_repository.bzl b/lib/private/symlink_repository.bzl
+new file mode 100644
+index 0000000..57c7857
+--- /dev/null
++++ b/lib/private/symlink_repository.bzl
+@@ -0,0 +1,19 @@
++"""
++Workaround for native repo rules not working in module extensions:
++https://groups.google.com/g/bazel-discuss/c/xpsg3mWQPZg
++"""
++
++def _symlink_repository_impl(repository_ctx):
++    relative_path = repository_ctx.attr.path
++    workspace_root = repository_ctx.path(Label("@//:MODULE.bazel")).dirname
```

Comment:
I wonder if you can have the workspace root of `aspect_bazel_lib` by changing this label to `"//:MODULE.bazel"`

Reply:
Tried that, CI still fails